### PR TITLE
Modernize typings

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Changed typings according to inertiajs/inertia#68

I think inertiajs/inertia first needs to be updated and the dependencies of this adapter should be updated accordingly, so will leave this as a draft until that's merged.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,51 +1,47 @@
...
-      key: React.Key
-      props: PageProps
-    }) => React.ReactNode
-    initialPage: Inertia.Page<PageProps>
-    resolveComponent: (
-      name: string
-    ) => React.ComponentType | Promise<React.ComponentType>
-    transformProps?: (props: PagePropsBeforeTransform) => PageProps
-  }>
+import * as Inertia from '@inertiajs/inertia'
```
Comment (sebastiandedeyne) [inline index.d.ts]: Seems to work fine without there. Can we get rid of them?

I've had issues in the past where adding `import` statements to decleration files made Webpack consider them as "real" scripts…

Comment (MichaelDeBoey) [reply index.d.ts]: WebStorm seems to complain about all the `Inertia.*` in the file then, but I can get rid of them by doing

```ts
import { PagePropsBeforeTransform as InertiaPagePropsBeforeTransform } from '@inertia/inertia'
import { FunctionComponent } from 'react'
```

If you want? 🤔

**Comment by MichaelDeBoey** [comment]
@sebastiandedeyne `@inertiajs/inertia` will first need to do a release I think, so I can update the dependency (with the new types) here? 🤔

File: index.d.ts (hunk 1 above)
Comment (sebastiandedeyne) [reply index.d.ts]: Hmm you're right. Keep it as is for now, we'll see if it has any negative effects.

**Comment by sebastiandedeyne** [comment]
Is this still a draft? Looks ready to me :)

**Comment by MichaelDeBoey** [comment]
@sebastiandedeyne Looks ready to be merged to me 🙂

**Pull request merged**